### PR TITLE
Api Facade and Comments

### PR DESCRIPTION
**Pull request by yura-kim**
Create an api facade python file and some temporary data based on our defined endpoints.
- Simple tests included with the api-facade, just run the file to see results of all functions

Create new urls.
- Go to http://127.0.0.1:5000/courses to view course list
- Click on swen-344 course to see students and grades
- Click on a grade to see comment(s)

Created a page for course detail view.

For #4 : Added collapsible comment section for each grade which lists grades in no particular order. I'll order them based on timestamp later.
I'll add a textbox to create a comment later.

### CODE REVIEW

**Comment by zsam56** [comment]
Awesome! looks good to me.

File: api_facade.py [hunk 1]
```diff
@@ -0,0 +1,290 @@
+courses = {
+    'swen-101':
+        {
+            "courseName": "Freshman Seminar",
+            "courseId": "swen-101",
+            "numGrades": 0,
+            "numStudents": 60,
+            "instructor": "0001",
+            "students": ["0001", "0002", "0003", "0004"],
+        },
```
Comment (ShaydeNofziger) [inline api_facade.py:10]: Just some small things here we may or may not want to address..

- All fields with relations are probably going to be to the foreign key / id of that entity. So we should probably follow that convention here so there's minimal rework for the differences in what the API will be giving us. 
- Also, some of these fields can be calculated based on the data and I don't think the API will be returning that directly, such as `numStudents` and `numGrades`. But we can get `numStudents` from the list of `students` and we can get `numGrades` by requesting all grades for a given `course.id`.

File: api_facade.py [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+    "0002":
+        {
+            "name": "Malachowsky",
+        },
+    "0003":
+        {
+            "name": "Vallino",
+        },
+}
```
Comment (ShaydeNofziger) [inline api_facade.py:126]: Similar to what was said above, the api will be expecting integers for the id's of the entities.

File: api_facade.py [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+        }
+}
+
+
+student_courses = {
+    "swen-3449999": {
+        "student_id": "9999",
+        "course_id": "swen-344",
+        "grade": "A",
+        "locked": 0,
```
Comment (ShaydeNofziger) [inline api_facade.py:154]: API will be returning `true` or `false`

File: api_facade.py [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+}
+
+
+comments = {
+    "0001":
+        {
+            "student_course_id": "swen-3449999",
+            "comment": "Thanks",
+            "timestamp": "Yesterday",
+            "author": "Lexie",
```
Comment (ShaydeNofziger) [inline api_facade.py:183]: Should this be a foreign key or id of a user? What if the authors share the same name?

**Review by ShaydeNofziger — COMMENTED** [review]
Just some minor things regarding what the values / types should be for the API responses, based on what the API will be returning. We want to align those as closely as possible so we have minimal rework once connected to the API.

File: api_facade.py (hunk 4 above)
Comment (yura-kim) [reply api_facade.py:183]: I was thinking about making it the id of a user, in which case we should have users include both instructors and students. But we need a way to differentiate between the two.

Comment (ShaydeNofziger) [reply api_facade.py:183]: Ah right, and our design doesn't currently support that....What if there was a boolean `isProf` to distinguish the two? I think that would work for our use cases and would be easy to add to the design.

**Review by ShaydeNofziger — APPROVED** [review]

**Pull request merged**